test/permitTwo/redeem.t.sol

```diff
@@ -0,0 +1,195 @@
+        uint256 currentDeadline = getFiveMinutesDeadlineFromNow();
+
+        ISignatureTransfer.PermitTransferFrom memory permit = ISignatureTransfer.PermitTransferFrom({
+            permitted: ISignatureTransfer.TokenPermissions({ token: archToken, amount: archTokenAmount }),
+            nonce: currentNonce,
+            deadline: currentDeadline
+        });
+
+        bytes32 msgToSign =
+            getRedeemWithPermit2MessageToSign(chainId, permit, address(gasworks), myRedeemData);
+        bytes memory signature = signMessage(ALICE_PRIVATE_KEY, msgToSign);
+
+        gasworks.redeemWithPermit2(permit, ALICE, signature, myRedeemData, false);
+
+        uint256 finalToTokenBalanceDiff = IERC20(toToken).balanceOf(ALICE) - previousToTokenBalance;
+
+        assertGe(finalToTokenBalanceDiff, adjustedMinReceiveAmount);
+        assertEq(previousArchTokenBalance - IERC20(archToken).balanceOf(ALICE), archTokenAmount);
+    }
+
+    /**
+     * Loads params and call instructions (quote) from a local json file, and then
+     * runs it to redeem a chamber
+     */
+    function runLocalRedeemQuoteTest(string memory fileName) public {
+        path = string.concat(root, fileName);
+        json = vm.readFile(path);
+        (
+            uint256 networkId,
+            uint256 blockNumber,
+            address archToken,
+            uint256 archTokenAmount,
+            address toToken,
+            uint256 minReceiveAmount,
+            ITradeIssuerV2.ContractCallInstruction[] memory contractCallInstructions
+        ) = parseRedeemQuoteFromJson(json);
+        redeemChamber(
+            networkId,
+            blockNumber,
+            archToken,
+            archTokenAmount,
+            toToken,
+            minReceiveAmount,
+            contractCallInstructions
+        );
+    }
+
+    /**
+     * Used to create json files, fetches a quote from arch and prints a JSON-readable
+     * quote in console, ready to be saved for new tests. The fork is needed to get the
+     * block number alongside the quote.
+     */
+    function printQuoteToCreateATest() public {
+        vm.createSelectFork("polygon");
+        fetchRedeemQuote(POLYGON_CHAIN_ID, POLYGON_AAGG, 30e18, POLYGON_USDT);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              SUCCESS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * [SUCCESS] Should redeem AAGG for USDT using permit2
+     */
+    function testRedeemWithPermit2FromAaggToUsdtOnPolygon() public {
+        // redeemChamber(POLYGON_CHAIN_ID, POLYGON_AAGG, POLYGON_USDT);
+        runLocalRedeemQuoteTest(
+            "/data/permitTwo/redeem/testRedeemWithPermit2FromAaggToUsdtOnPolygon.json"
+        );
+    }
+
+    /**
+     * [SUCCESS] Should redeem AMOD for USDT using permit2 [Test when supply is available]
+     */
+    // function testRedeemWithPermit2FromAmodToUsdtOnPolygon() public {
```

Comment:
Este test igual lo podrias hacer o no? solo que juegas con montos mas chicos ya que el supply es chico chico o que piensas

Reply:
It worked